### ISSUE TITLE
Wrong float data with groupby/mean

### ISSUE THREAD

**Issue by aguillot** [pola-rs/polars#8504]
### Polars version checks

- [X] I have checked that this issue has not already been reported.

- [X] I have confirmed this bug exists on the [latest version](https://pypi.org/project/polars/) of Polars.

### Issue description

With the latest version of polars, but first seen in 0.17.7

As a float, instead of a valid timestamp some rows become a much smaller value.

```
polars 0.17.9
┌────────────────────┬───────┬─────────────┐
│ visit_id           ┆ ide   ┆ timestamp   │
│ ---                ┆ ---   ┆ ---         │
│ str                ┆ i32   ┆ f64         │
╞════════════════════╪═══════╪═════════════╡
│ f2a3dd265266#195   ┆ -6851 ┆ 8.2279e-315 │
│ f2a3dd265266#195   ┆ -6850 ┆ 8.2279e-315 │
│ 5066bd8dc9cf#111   ┆ -6849 ┆ 8.2279e-315 │
│ f2a3dd265266#195   ┆ -6847 ┆ 8.2279e-315 │
│ …                  ┆ …     ┆ …           │
│ 26aa181f1f8c#95876 ┆ 1736  ┆ 1.6654e9    │
│ 08d90fe50c68#95874 ┆ 1739  ┆ 8.2284e-315 │
│ 8de29b660ac1#94368 ┆ 1744  ┆ 8.2284e-315 │
│ 26c0dab56fe7#21958 ┆ 1777  ┆ 1.6654e9    │
└────────────────────┴───────┴─────────────┘
```

```
polars 0.17.1
┌────────────────────┬───────┬───────────┐
│ visit_id           ┆ ide   ┆ timestamp │
│ ---                ┆ ---   ┆ ---       │
│ str                ┆ i32   ┆ f64       │
╞════════════════════╪═══════╪═══════════╡
│ f2a3dd265266#195   ┆ -6851 ┆ 1.6654e9  │
│ f2a3dd265266#195   ┆ -6850 ┆ 1.6654e9  │
│ 5066bd8dc9cf#111   ┆ -6849 ┆ 1.6654e9  │
│ f2a3dd265266#195   ┆ -6847 ┆ 1.6654e9  │
│ …                  ┆ …     ┆ …         │
│ 26aa181f1f8c#95876 ┆ 1736  ┆ 1.6654e9  │
│ 08d90fe50c68#95874 ┆ 1739  ┆ 1.6654e9  │
│ 8de29b660ac1#94368 ┆ 1744  ┆ 1.6654e9  │
│ 26c0dab56fe7#21958 ┆ 1777  ┆ 1.6654e9  │
└────────────────────┴───────┴───────────┘
```

### Reproducible example

```python
import polars as pl

df = pl.read_parquet("bug_data.parquet")

df = df.groupby(["visit_id", "ide"]).agg(
    [pl.mean("timestamp")]
)

assert df.get_column("timestamp").min() > 1
```

### Expected behavior

On polars version 0.17.1, all timestamps are valid

### Installed versions

<details>

```
---Version info---
Polars: 0.17.9
Index type: UInt32
Platform: Linux-5.15.0-70-generic-x86_64-with-glibc2.35
Python: 3.11.3 (main, Apr 19 2023, 18:26:08) [GCC 11.3.0]
---Optional dependencies---
numpy: <not installed>
pandas: <not installed>
pyarrow: <not installed>
connectorx: <not installed>
deltalake: <not installed>
fsspec: <not installed>
matplotlib: <not installed>
xlsx2csv: <not installed>
xlsxwriter: <not installed>
```

</details>

**Comment by aguillot** [comment]
[bug_data.parquet.zip](https://github.com/pola-rs/polars/files/11324086/bug_data.parquet.zip)

**Comment by aguillot** [comment]
The operation influencing the bug is mean. In the attached parquet file, all timestamps are valid integers.
After the groupby/mean operation, with polars 0.17.9, these timestamps (and other columns) have bad values.

```python
# Correct value in polars 0.17.1
>>> df.groupby(["visit_id","ide"]).mean().filter(pl.col("visit_id") == "ef0cb279ee49#32322")
shape: (1, 7)
┌────────────────────┬───────┬───────────┬───────┬────────┬───────┬─────┐
│ visit_id           ┆ ide   ┆ timestamp ┆ rssi  ┆ unique ┆ count ┆ ct  │
│ ---                ┆ ---   ┆ ---       ┆ ---   ┆ ---    ┆ ---   ┆ --- │
│ str                ┆ i32   ┆ f64       ┆ f64   ┆ f64    ┆ f64   ┆ f64 │
╞════════════════════╪═══════╪═══════════╪═══════╪════════╪═══════╪═════╡
│ ef0cb279ee49#32322 ┆ -2679 ┆ 1.6654e9  ┆ -61.0 ┆ 2.0    ┆ 2.0   ┆ 3.0 │
└────────────────────┴───────┴───────────┴───────┴────────┴───────┴─────┘


# Bad values in polars 0.17.9
>>> df.groupby(["visit_id","ide"]).mean().filter(pl.col("visit_id") == "ef0cb279ee49#32322")
shape: (1, 7)
┌────────────────────┬───────┬─────────────┬───────┬────────┬───────┬───────┐
│ visit_id           ┆ ide   ┆ timestamp   ┆ rssi  ┆ unique ┆ count ┆ ct    │
│ ---                ┆ ---   ┆ ---         ┆ ---   ┆ ---    ┆ ---   ┆ ---   │
│ str                ┆ i32   ┆ f64         ┆ f64   ┆ f64    ┆ f64   ┆ f64   │
╞════════════════════╪═══════╪═════════════╪═══════╪════════╪═══════╪═══════╡
│ ef0cb279ee49#32322 ┆ -2679 ┆ 8.2281e-315 ┆ -61.0 ┆ -61.0  ┆ -61.0 ┆ -61.0 │
└────────────────────┴───────┴─────────────┴───────┴────────┴───────┴───────┘
```

**Comment by aguillot** [comment]
fixed in version 0.17.10
